datafusion/core/src/datasource/stream.rs

```diff
@@ -103,19 +105,46 @@ impl FromStr for StreamEncoding {
     }
 }
 
-/// The configuration for a [`StreamTable`]
+/// The StreamProvider trait is used as a generic interface for reading and writing from streaming
+/// data sources (such as FIFO, Websocket, Kafka, etc.).  Implementations of the provider are
+/// responsible for providing a `RecordBatchReader` and optionally a `RecordBatchWriter`.
+pub trait StreamProvider: std::fmt::Debug + Send + Sync {
+    /// Get a reference to the schema for this stream
+    fn schema(&self) -> SchemaRef;
+    /// Needed for `PartitionStream` - maybe there is a better way to do this.
```

Comment:
I think changing `schema` to return `&SchemaRef` would be enough -- then anyone who needed an owned `SchemaRef` could clone the result. I don't think we need separate methods